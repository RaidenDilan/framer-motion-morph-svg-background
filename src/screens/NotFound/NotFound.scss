@import '../../theme/colors.module.scss';
@import '../../theme/mixins.module.scss';

.NotFound {
  @include flex(center, center, true);
  min-height: calc(100vh - 8em);
  backdrop-filter: blur(6px) grayscale(25%);

  h1 {
    color: $White;
    font-size: 5em;
  }
}