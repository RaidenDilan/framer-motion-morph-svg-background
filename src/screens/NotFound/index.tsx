import React from 'react';
import './NotFound.scss';

const NotFound = () => {
  return (
    <div className="NotFound">
      <h1>404</h1>
    </div>
  );
};

export default NotFound;
