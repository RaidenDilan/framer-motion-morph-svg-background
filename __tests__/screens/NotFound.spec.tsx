import React from 'react';
import { render } from '@testing-library/react';

import NotFound from '../../src/screens/NotFound';

const navigate = jest.fn();

jest.mock('react-router-dom', () => {
  const requireActual = jest.requireActual('react-router-dom');
  return {
    ...requireActual,
    useNavigate: () => navigate,
  };
});

describe(__filename, () => {
  it('should render without crashing', () => {
    const { asFragment } = render(<NotFound />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the correct text', () => {
    const wrapper = render(<NotFound />);
    expect(wrapper.getByText('404')).toBeInTheDocument();
  });
});
