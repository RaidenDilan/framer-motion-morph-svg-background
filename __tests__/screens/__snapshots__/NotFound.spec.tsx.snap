// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Users/Raiden/development/.boilerplates/vite-boilerplate/__tests__/screens/NotFound.spec.tsx should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="NotFound"
  >
    <h1>
      404
    </h1>
  </div>
</DocumentFragment>
`;
